### PR TITLE
design spec for NVMeoF support

### PR DESCRIPTION
**Pull request by qwren**
This patch is about the implementation for NVMeoF in OpenSDS.

Signed-off-by: Qiaowei Ren <qiaowei.ren@intel.com>

### CODE REVIEW

File: specs/capri/nvmeof_support.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# NVMeoF support
+
+**Authors:** [Intel](https://github.com/qwren)
+
+This is the design spec for the NVMeoF feature.
+
+## Motivation
+
+The NVMe over Fabrics (NVMeOF) is an emerging technology, and it gives data centers unprecedented access to NVMe SSD storage. Like iSCSI, NVMeОF is a network protocol used to communicate between a host and a storage system over a network (aka fabric). Ompared with iSCSI, NVMeOF has much lower latency, in practice adding just a few microseconds to cross the network. This makes the difference between local storage and remote storage very small.
+
```
Comment (xing-yang) [inline specs/capri/nvmeof_support.md]: s/Ompared/Compared

File: specs/capri/nvmeof_support.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
++}
++
++func (nof *Nvmeof) Detach(conn map[string]interface{}) error {
++        NvmeofCon := ParseNvmeofConnectInfo(conn)
++
++        return DisConnect(NvmeofCon.Nqn)
++}
+```
+
+### NVMeoF support for Native OpenSDS Southbound LVM drive
```
Comment (xing-yang) [inline specs/capri/nvmeof_support.md]: s/drive/driver

File: specs/capri/nvmeof_support.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+type NvmeofTarget interface {
+        CreateNvmeofTarget(volId, tgtNqn, path, hostIp, initiator string, chapAuth []string) error
+        GetNvmeofTarget(nqn string) int
+        RemoveNvmeofTarget(volId, nqn string) error
+}
+```
+
+### Ansible impact
+
+RDMA Ethernet card is necessary for real NVMeoF Initiator/Target environment. Hardware environment have to be configured in Ansible, and it is necessary to load the following modules to make use of RDMA feature in userspace:
```
Comment (xing-yang) [inline specs/capri/nvmeof_support.md]: s/have to be/has to be

File: specs/capri/nvmeof_support.md (hunk 2 above)
Comment (qwren) [reply specs/capri/nvmeof_support.md]: done

**Comment by leonwanghui** [comment]
@xing-yang I think this design spec could be merged.

**Comment by xing-yang** [comment]
lgtm

**Pull request merged**